### PR TITLE
Sort results by severity

### LINKED ISSUE #8: Sort by severity first?
Sorting by line number makes a lot of sense, but I ignore most warnings anyway (or rather, most warnings I see are ones I evaluated and decided to ignore at some point in the past).

I think it might make sense to put the higher severity `error` at the bottom of the list (since we are scrolling up in our terminal to view it anyways.

Also, it might make sense to reorder the files so those with only warnings and no errors come first. 

This gets that one error you missed down at the bottom of the output every time.

### PR DESCRIPTION
**Pull request by kevva**
I couldn't think of a cleaner way to solve this.

Fixes #8.

### CODE REVIEW

**Comment by sindresorhus** [comment]
Easier review: https://github.com/sindresorhus/eslint-formatter-pretty/pull/12/files?w=1

File: index.js [hunk 1]
```diff
@@ -16,60 +16,84 @@ module.exports = function (results) {
...
-
-		if (lines.length !== 0) {
-			lines.push({type: 'separator'});
-		}
+			if (a.errorCount > 0 && b.errorCount === 0) {
+				return 1;
+			}
 
-		var filePath = result.filePath;
+			return 0;
```
Comment (sindresorhus) [inline index.js]: Why not just:

```js
return b.errorCount - a.errorCount;
```

File: index.js [hunk 2]
```diff
@@ -16,60 +16,84 @@ module.exports = function (results) {
...
+				firstLineCol: messages[0].line + ':' + messages[0].column
 			});
+
+			messages
+				.sort(function (a, b) {
+					if ((a.fatal || a.severity === 2) && (!b.fatal || b.severity !== 2)) {
+						return 1;
+					}
+
+					if ((!a.fatal || a.severity !== 2) && (b.fatal || b.severity === 2)) {
```
Comment (sindresorhus) [inline index.js]: `(!a.fatal || a.severity !== 2)` => `!(a.fatal || a.severity === 2)`?

You could then put it in a variable and just reuse with both checks, but negate accordingly.

**Comment by sindresorhus** [comment]
@kevva A couple of inline comments, but generally looks good.

@jamestalmage ?

**Comment by kevva** [comment]
@sindresorhus, phew, forgot about this. Fixed the inline comments and rebased against `master`.

**Comment by sindresorhus** [comment]
screen shot 2016-09-16 at 15 11 43

**Comment by kevva** [comment]
Awesome! 🎈

**Pull request merged**